org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/CompletionEngine.java

```diff
@@ -3689,7 +3689,13 @@ private void completionOnQualifiedAllocationExpression(ASTNode astNode, Binding
 					false);
 			}
 		}
-		findCompletionsForArgumentPosition(constructorsFound, argTypes != null ? argTypes.length : 0, scope);
+		// here we assume that if first type is null we have a <CompleteOnName:>
```

Comment:
Sorry, I'm not knowledgeable of the completion engine. Can you explain what is `CompleteOnName:`?

Reply:
This seems to work OK, but I wonder, why we should use such indirect reasoning. Could this break if a provided first argument is not resolvable? What would be the `resolvedType` in that case, `null` or some `Problem(Reference)Binding`?

Alternatively we could explicitly check if `allocExpression.arguments[0]` is the completion node. I'm just surprised we don't seem to have a simple query that would answer this, but that could of course be added right now.

Reply:
@stephan-herrmann This is how the parser.assistNode looks like

`CompletionOnQualifiedAllocationExpression@103 "<CompleteOnAllocationExpression:new PersonDetails(<CompleteOnMessageSendName:()>, 0)>"`

I also checked when the first argument is not resolvable, then it will be null as well. But we only go through here for the first argument of the constructor since we want to also suggest constructor signature as completion items which is used for signature help in jdt ui and jdt.ls.

We cannot use `allocExpression.arguments[0]` since it will not be the actual completion node as I have shown above. 

Now since we don't hit this method other than for the first parameter, I wonder why we really need to pass the index based on arguments, we could just pass 0. WDYT ?

Reply:
@stephan-herrmann could you have a look at my last comment and let me know if we need to change the fix or if this is ok for merging ?

Reply:
Relying on `argTypes[0] == null` depends on how `SelectionOnSingleNameReference.resolveType()` at all are implemented. Incidentally, in #1077 I'm working on a change that _could_ indeed cause `resolveType()` to invoke super and thus set a non-null `resolvedType`. I haven't checked if that is a real or only theoretical conflict between both changes.

> We cannot use allocExpression.arguments[0] since it will not be the actual completion node as I have shown above.


For completeness, here's a sketch of what I was thinking of:
```
	if (arguments != null && arguments.length > 0 && arguments[0] instanceof CompletionOnSingleNameReference) {
		completedArgLen = 0; // just confirming :)
	} else if (argTypes != null) {
		completedArgLen = argTypes.length;
	}
```
Tests pass, but of course the instanceof check is incomplete, might be improved e.g., by a marker interface AnyCompletionNode or similar approach.

This would formalize the case were indeed we are completing a first argument.

But maybe all this is overkill.

> Now since we don't hit this method other than for the first parameter, I wonder why we really need to pass the index based on arguments, we could just pass 0. WDYT ?

That adds another indirect reasoning. Why should we be sure that it's only for the first parameter? Class header comment in `CompletionOnQualifiedAllocationExpression` says differently, no?